lab_3_generate_by_ngrams/main.py

```diff
@@ -41,6 +45,19 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or not text:
+            return None
+        tokens = []
+        for token in text.lower():
+            if token.isspace() and tokens[-1] != self._end_of_word_token:
+                tokens.append(self._end_of_word_token)
+            elif token.isalpha():
+                tokens.append(token)
+        if not text[-1].isalnum():
+            tokens.append(self._end_of_word_token)
+        if tokens.count(self._end_of_word_token) == len(tokens):
```

Comment:
i will need an explanation

Reply:
I meant that if text does not contain any valid tokens we return None, but now I think maybe it could be done easier by isalpha()...